compiler/rustc_passes/src/lang_items.rs

```diff
@@ -180,6 +181,83 @@ impl LanguageItemCollector<'tcx> {
             self.items.groups[group as usize].push(item_def_id);
         }
     }
+
+    fn collect_item_extended(&mut self, item_index: usize, hir_id: HirId, span: Span) {
+        let item_def_id = self.tcx.hir().local_def_id(hir_id).to_def_id();
+        let lang_item = LangItem::from_u32(item_index as u32).unwrap();
+        let name = lang_item.name();
+
+        self.collect_item(item_index, item_def_id);
+
+        // Now check whether the lang_item has the expected number of generic
+        // arguments. Binary and indexing operations have one (for the RHS/index),
+        // unary operations have no generic arguments.
+
+        let expected_num = match lang_item {
+            LangItem::Add
+            | LangItem::Sub
+            | LangItem::Mul
+            | LangItem::Div
+            | LangItem::Rem
+            | LangItem::BitXor
+            | LangItem::BitAnd
+            | LangItem::BitOr
+            | LangItem::Shl
+            | LangItem::Shr
+            | LangItem::AddAssign
+            | LangItem::SubAssign
+            | LangItem::MulAssign
+            | LangItem::DivAssign
+            | LangItem::RemAssign
+            | LangItem::BitXorAssign
+            | LangItem::BitAndAssign
+            | LangItem::BitOrAssign
+            | LangItem::ShlAssign
+            | LangItem::ShrAssign
+            | LangItem::Index
+            | LangItem::IndexMut => Some(1),
+
+            LangItem::Neg | LangItem::Not | LangItem::Deref | LangItem::DerefMut => Some(0),
+
+            // FIXME: add more cases?
```

Comment:
How difficult would it be to go through all the lang items and check all of them have been accounted for?

Reply:
I've added all the other lang items that are traits now. The code below assumes that it is looking at a trait, so adding the non-trait lang items would be more work, and also less important, because e.g. the `bool` lang item is an `impl`, and writing `impl<T> bool<T> { ... }` already gives you an error from elsewhere in the compiler.